addons/hr_holidays/models/hr_leave.py

```diff
@@ -312,9 +320,15 @@ def _search_description(self, operator, value):
         return [('id', 'in', leaves.ids)]
 
     @api.depends('holiday_status_id')
-    def _compute_state(self):
+    def _compute_from_holiday_status_id(self):
         for holiday in self:
             holiday.state = 'confirm' if holiday.validation_type != 'no_validation' else 'draft'
+            if holiday.holiday_status_id.requires_allocation == 'yes':
+                holiday.holiday_allocation_id = self.env['hr.leave.allocation'].search([('holiday_status_id', '=', holiday.holiday_status_id.id),
```

Comment:
We should perform a read_group here. Otherwise we call the database for each record.